modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/OperationMessageProcessor.java

```diff
@@ -89,28 +108,72 @@
 
   private ExecutionMediator executionMediator;
   private OperationExecutor operationExecutor;
+  private PolicyManager policyManager;
   protected ReturnDelegate returnDelegate;
 
   public OperationMessageProcessor(ExtensionModel extensionModel,
                                    OperationModel operationModel,
                                    ConfigurationProvider configurationProvider,
                                    String target,
                                    ResolverSet resolverSet,
-                                   ExtensionManagerAdapter extensionManager) {
+                                   ExtensionManagerAdapter extensionManager,
+                                   PolicyManager policyManager) {
     super(extensionModel, operationModel, configurationProvider, extensionManager);
     this.extensionModel = extensionModel;
     this.operationModel = operationModel;
     this.resolverSet = resolverSet;
     this.target = target;
     this.entityMetadataMediator = new EntityMetadataMediator(operationModel);
+    this.policyManager = policyManager;
   }
 
   @Override
   public Event process(Event event) throws MuleException {
-    return (Event) withContextClassLoader(getExtensionClassLoader(), () -> {
+    return withContextClassLoader(getExtensionClassLoader(), () -> {
       Optional<ConfigurationInstance> configuration = getConfiguration(event);
-      ExecutionContextAdapter operationContext = createExecutionContext(configuration, event);
-      return doProcess(event, operationContext);
+
+      ComponentIdentifier operationIdentifier =
```

Comment:
I think you are right. It would be good if we can have a ComponentIdentiifer ComponentModel.getIdentifier() method. The problem is that the ComponentModel does not have a relationship with its ExtensionModel so it requires some changes to be able to add such method.